libbpfgo.go

```diff
@@ -239,23 +270,35 @@ const (
 	Tracing
 	XDP
 	Cgroup
+	CgroupLegacy
 	Netns
 )
 
 type BPFLink struct {
-	link      *C.struct_bpf_link
-	prog      *BPFProg
-	linkType  LinkType
-	eventName string
+	link       *C.struct_bpf_link
+	prog       *BPFProg
+	linkType   LinkType
+	eventName  string
+	attachType BPFAttachType // needed by legacy cgroup attachment
+	cgroupDir  string        // needed by legacy cgroup attachment
+}
+
+func (l *BPFLink) DestroyLegacy(linkType LinkType) error {
+	switch l.linkType {
+	case CgroupLegacy:
+		return l.prog.DetachCgroupLegacy(l.cgroupDir, l.attachType)
+	}
+	return fmt.Errorf("unable to destroy legacy link")
 }
 
 func (l *BPFLink) Destroy() error {
-	ret := C.bpf_link__destroy(l.link)
-	if ret < 0 {
+	if l.link == nil { // might be a legacy/fake BPFLink
```

Comment:
Should we explicitly mark links as being created via this legacy api so as to better surface errors?

Reply:
Doing so in the version about to be pushed.